test/pytest/test_sr.py

```diff
@@ -0,0 +1,73 @@
+import warnings
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+import hls4ml
+
+test_root_path = Path(__file__).parent
+
+
+@pytest.fixture(scope='module')
+def data():
+    X = 2 * np.random.rand(100, 5)
+    y = 2.5382 * np.cos(X[:, 3]) + X[:, 0] ** 2 - 0.5
+
+    return X, y
+
+
+def test_hlssr(data):
+    expr = 'x0**2 + 2.5382*cos_lut(x3) - 0.5'
+
+    lut_functions = {'cos_lut': {'math_func': 'cos', 'range_start': -4, 'range_end': 4, 'table_size': 2048}}
+
+    output_dir = str(test_root_path / 'hls4mlprj_sr')
+
+    hls_model = hls4ml.converters.convert_from_symbolic_expression(
+        expr,
+        n_symbols=5,
+        precision='ap_fixed<18,6>',
+        output_dir=output_dir,
+        lut_functions=lut_functions,
+        hls_include_path='',
+        hls_libs_path='',
+    )
+    hls_model.write()
+    hls_model.compile()
+
+    X, y = data
+    y_hls = hls_model.predict(X)
+    y_hls = y_hls.reshape(y.shape)
+
+    np.testing.assert_allclose(y, y_hls, rtol=1e-2, atol=1e-2, verbose=True)
+
+
+def test_pysr_luts(data):
+    try:
+        from pysr import PySRRegressor
+    except ImportError:
+        warnings.warn(UserWarning('Failed to import PySR, test will be skipped.'), stacklevel=1)
```

Comment:
I am worried that this makes this test too easy to ignore.

Reply:
Note in the pytest:

```
=============================== warnings summary ===============================
test/pytest/test_sr.py::test_pysr_luts
  /builds/fastmachinelearning/hls4ml/test/pytest/test_sr.py:50: UserWarning: Failed to import PySR, test will be skipped.
    warnings.warn(UserWarning('Failed to import PySR, test will be skipped.'), stacklevel=1)
test/pytest/test_sr.py::test_pysr_luts
  /home/hls4ml/miniconda/envs/hls4ml-testing/lib/python3.8/site-packages/_pytest/python.py:199: PytestReturnNotNoneWarning: Expected None, but test/pytest/test_sr.py::test_pysr_luts returned True, which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
    warnings.warn(
```

Reply:
The alternative is to have a full installation of PySR. This can be tricky, since we would need to update the underlying docker image to have it (and Julia and all dependencies). Furthermore, this test actually takes time to produce meaningful results, it's basically training. And we need to do it, since we would ideally have to check if the result uses the LUTs we supplied (an error in the implementation may make the function output nonsense, thus making PySR avoid using it)

as for the warning, that's a good catch, i'll fix the test.

Reply:
I changed the test to skip it if pysr cannot be imported. There's a nice feature of pytest specifically for these things, I didn't know about it. Now there's a clean message: `SKIPPED [1] test_sr.py:49: Failed to import PySR, test will be skipped.` which doesn't affect the overall outcome, but still flags the test in a special way in the final report.

Reply:
Do we want to change the setup so that it's not skipped in our standard testing? But that can be in a different PR. I'll look more closely tomorrow since today is a holiday.

Reply:
If we don't want it skipped, we need to change the environment of the container to include this dependency, which IMO should be a separate PR.

Reply:
I would recommend making a separate PR to change the environment, but for this PR, this is good.